yb-voyager/cmd/exportData.go

```diff
@@ -788,14 +858,262 @@ func getInitialTableList() (map[string]string, []sqlname.NameTuple) {
+	if err != nil {
+		return nil, nil, fmt.Errorf("error getting name registry list: %v", err)
+	}
+
+	rootTables := make([]sqlname.NameTuple, 0)
+	for _, v := range partitionsToRootTableMap {
+		tuple, err := namereg.NameReg.LookupTableName(v)
+		if err != nil {
+			utils.ErrExit("look up failed for the table name: %s: %v", v, err)
+		}
+		rootTables = append(rootTables, tuple)
+	}
+	rootTables = lo.UniqBy(rootTables, func(t sqlname.NameTuple) string {
+		return t.ForKey()
+	})
+
+	// Finding all the partitions of all root tables part of migration, and report if there any new partitions added
+	rootToNewLeafTablesMap, err := detectAndReportNewLeafPartitionsOnPartitionedTables(rootTables, registeredList)
+	if err != nil {
+		return nil, nil, fmt.Errorf("detecting new leaf tables on the partitioned tables: %v", err)
+	}
+
+	firstRunTableWithLeafParititons, currentRunTableListWithLeafPartitions, err := applyTableListFlagsOnCurrentAndRemoveRootsFromBothLists(registeredList, source.TableList, source.ExcludeTableList, rootToNewLeafTablesMap, rootTables, firstRunTableWithLeafsAndRoots)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error applying table list flags for current list and remove roots: %v", err)
+	}
+
+	//Reporting the guardrail msgs only on leaf tables to be consistent so filtering the root table from both the list
+	guardrailsAroundFirstRunAndCurrentRunTableList(firstRunTableWithLeafParititons, currentRunTableListWithLeafPartitions)
+
+	return partitionsToRootTableMap, firstRunTableWithLeafsAndRoots, nil
+
+}
+
+func applyTableListFlagsOnCurrentAndRemoveRootsFromBothLists(
+	registeredList []sqlname.NameTuple,
+	tableListViaFlag string,
+	excludeTableListViaFlag string,
+	rootToNewLeafTablesMap map[string][]string,
+	rootTables []sqlname.NameTuple,
+	firstRunTableWithLeafsAndRoots []sqlname.NameTuple) ([]sqlname.NameTuple, []sqlname.NameTuple, error) {
+
+	//apply include/exclude flags and if a new table is passed (which is not present in name registry), then error out Unknown table
+	currentRunTableListFilteredViaFlags, err := applyTableListFlagsOnFullListAndAddLeafPartitions(registeredList, tableListViaFlag, excludeTableListViaFlag)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error in apply table list filter on registered list for the flags in current run: %v", err)
+	}
+	//checks if a given table is new leaf table or not
+	isNewLeafTable := func(t sqlname.NameTuple) bool {
+		for _, leafs := range rootToNewLeafTablesMap {
+			if slices.Contains(leafs, t.AsQualifiedCatalogName()) {
+				return true
+			}
+		}
+		return false
+	}
+
+	var currentRunTableListWithLeafPartitions []sqlname.NameTuple
+	for _, t := range currentRunTableListFilteredViaFlags {
+		if lo.ContainsBy(rootTables, func(r sqlname.NameTuple) bool {
+			return t.Equals(r)
+		}) {
+
+			//Remove root table
+			continue
+		}
+
+		//Filtering the new leaf tables if present in filteredListWithoutRootTable as we have reported above
+		if isNewLeafTable(t) {
```

Comment:
Why/how would the new leaf appear in currentRunTableListFilteredViaFlags?

Reply:
While applying include/exclude flags, we also add leaf partitions in case only the root is passed in flags.